libnopegl/src/backends/gl/glcontext.c

```diff
@@ -341,6 +341,12 @@ static int glcontext_probe_settings(struct glcontext *glcontext)
     struct gpu_limits *limits = &glcontext->limits;
 
     GET(GL_MAX_VERTEX_ATTRIBS, &limits->max_vertex_attributes);
+    /*
+     * macOS and iOS OpenGL drivers pass gl_VertexID and gl_InstanceID as
+     * standard attributes and forget to count them in GL_MAX_VERTEX_ATTRIBS.
+     */
+    if (glcontext->platform == NGL_PLATFORM_MACOS || glcontext->platform == NGL_PLATFORM_IOS)
+        limits->max_vertex_attributes -= 2;
```

Comment:
lol.

Do you have a source for that or that's an observation you made by experimenting?

Reply:
Experimentation + also confirmed by https://projects.blender.org/blender/blender/commit/c971e812d5e2e5c45ed262ccd3bccfcf547b9ff7